### PR TITLE
Implement power increase for STAB moves on Ogerpon masks

### PR DESCRIPTION
**Pull request by Elitemagikarp**
Research thread here: https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/post-9780649

### CODE REVIEW

**Comment by Karthik99999** [comment]
A couple things:
- It's not just STABs that get boosted. Multiple people have found that all moves seem to be getting some sort of boost.
- We do not have concrete evidence that this is actually a 1.2x base power boost yet.

File: data/items.ts [hunk 1]
```diff
@@ -1101,6 +1101,12 @@ export const Items: {[itemid: string]: ItemData} = {
 		name: "Cornerstone Mask",
 		spritenum: 758,
 		forcedForme: "Ogerpon-Cornerstone",
+		onBasePowerPriority: 15,
+		onBasePower(basePower, user, target, move) {
+			if (user.baseSpecies.num === 1017 && (move.type === 'Grass' || move.type === 'Rock')) {
```
Comment (Karthik99999) [inline data/items.ts:1106]:
```suggestion
			if (user.baseSpecies.name.startsWith('Ogerpon-Cornerstone')) {
```

File: data/items.ts [hunk 2]
```diff
@@ -2506,6 +2512,12 @@ export const Items: {[itemid: string]: ItemData} = {
 		name: "Hearthflame Mask",
 		spritenum: 760,
 		forcedForme: "Ogerpon-Hearthflame",
+		onBasePowerPriority: 15,
+		onBasePower(basePower, user, target, move) {
+			if (user.baseSpecies.num === 1017 && (move.type === 'Grass' || move.type === 'Fire')) {
```
Comment (Karthik99999) [inline data/items.ts:2517]:
```suggestion
			if (user.baseSpecies.name.startsWith('Ogerpon-Hearthflame')) {
```

File: data/items.ts [hunk 3]
```diff
@@ -7107,6 +7119,12 @@ export const Items: {[itemid: string]: ItemData} = {
 		name: "Wellspring Mask",
 		spritenum: 759,
 		forcedForme: "Ogerpon-Wellspring",
+		onBasePowerPriority: 15,
+		onBasePower(basePower, user, target, move) {
+			if (user.baseSpecies.num === 1017 && (move.type === 'Grass' || move.type === 'Water')) {
```
Comment (Karthik99999) [inline data/items.ts:7124]:
```suggestion
			if (user.baseSpecies.name.startsWith('Ogerpon-Wellspring')) {
```

**Review by Karthik99999 — CHANGES_REQUESTED** [review]

File: data/items.ts [hunk 4]
```diff
@@ -1101,6 +1101,12 @@ export const Items: {[itemid: string]: ItemData} = {
 		name: "Cornerstone Mask",
 		spritenum: 758,
 		forcedForme: "Ogerpon-Cornerstone",
+		onBasePowerPriority: 15,
+		onBasePower(basePower, user, target, move) {
+			if (user.baseSpecies.name.startsWith('Ogerpon-Cornerstone')) {
```
Comment (DaWoblefet) [inline data/items.ts:1106]: `if (user.baseSpecies.name === 'Ogerpon-Cornerstone') {`

[the remark this answers is not in the document]
Comment (DaWoblefet) [reply data/items.ts:1106]: As Marty pointed out, I forgot the Tera formes have different names. Ignore this.

**Comment by DaWoblefet** [comment]
Thank you!

**Pull request merged**